File: online-assignment/src/app/core/model/system/response-wrapper.ts
export declare class ResponseWrapper<T> {
  static SUCCESS: number;
  static ERROR: number;
  static WARNING: number;
  apiVersion: string;
  status: ResponseWrapperStatus;
  error: ResponseWrapperError;
  data: Array<T>;
}
export declare class ResponseWrapperError {
  code: number;
  message: string;
  details: string;
  errors: Array<string>;
}
export declare class ResponseWrapperStatus {
  code: number;
  message: string;
}

File: online-assignment-web-base/src/main/java/com/elearning/onlineassignment/models/ResponseWrapper.java
package com.elearning.onlineassignment.models;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class ResponseWrapper<T>
{

    private String version;

    private ResponseWrapperStatus status;

    private ResponseWrapperError error;

    private List<T> data;

    public ResponseWrapper()
    {
    }

    public void setStatus( ResponseWrapperStatus status )
    {
        this.status = status;
    }

    public void setError( ResponseWrapperError error )
    {
        this.error = error;
    }

    public ResponseWrapperStatus getStatus()
    {
        return status;
    }

    public ResponseWrapperError getError()
    {
        return error;
    }

    public List<T> getData()
    {
        return data;
    }

    public void setData( List<T> data )
    {
        this.data = data;
    }

    public String getVersion()
    {
        return version;
    }

    public void setVersion( String version )
    {
        this.version = version;
    }
}
